spikes/aks-secure-baseline/cluster-manifests/cluster-baseline-settings/flux.yaml

```diff
@@ -0,0 +1,183 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    app.kubernetes.io/name: flux
+  name: flux
+  namespace: cluster-baseline-settings
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: flux
+  labels:
+    app.kubernetes.io/name: flux
+rules:
+  - apiGroups: ['*']
+    resources: ['*']
+    verbs: ['*']
+  - nonResourceURLs: ['*']
+    verbs: ['*']
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: flux
+  labels:
+    app.kubernetes.io/name: flux
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: flux
+subjects:
+  - kind: ServiceAccount
+    name: flux
+    namespace: cluster-baseline-settings
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: flux
+  namespace: cluster-baseline-settings
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: flux
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      annotations:
+        prometheus.io/port: "3031"
+      labels:
+        app.kubernetes.io/name: flux
+    spec:
+      nodeSelector:
+        kubernetes.io/os: linux
+        agentpool: npuser01
+      serviceAccountName: flux
+      volumes:
+      - name: git-key
+        secret:
+          secretName: flux-git-deploy
+          defaultMode: 0400
+      containers:
+      - name: flux
+        # PRODUCTION READINESS CHANGE REQUIRED
+        # This image should be sourced from a non-public container registry, such as the
+        # one deployed along side of this reference implementation.
+        # az acr import --source docker.io/fluxcd/flux:1.21.1 -n <your-acr-instance-name>
+        # and then set this to
+        # image: <your-acr-instance-name>.azurecr.io/fluxcd/flux:1.21.1
+        image: docker.io/fluxcd/flux:1.21.1
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          capabilities:
+            drop:
+            - ALL
+          allowPrivilegeEscalation: false
+          # create folder in the root fs when cloning repos
+          readOnlyRootFilesystem: false
+          # access to root folder like /.kube/config
+          runAsNonRoot: false
+        volumeMounts:
+        - name: git-key
+          mountPath: /etc/fluxd/ssh
+          readOnly: true
+        resources:
+          requests:
+            cpu: 50m
+            memory: 64Mi
+        ports:
+        - containerPort: 3030
+        livenessProbe:
+          httpGet:
+            port: 3030
+            path: /api/flux/v6/identity.pub
+          initialDelaySeconds: 5
+          timeoutSeconds: 5
+        readinessProbe:
+          httpGet:
+            port: 3030
+            path: /api/flux/v6/identity.pub
+          initialDelaySeconds: 5
+          timeoutSeconds: 5
+        args:
+        - --git-url=https://github.com/retaildevcrews/ngsa.git
```

Comment:
This file was copied from aks-secure-baseline and modified. The only changes here is to point flux to the spikes folder instead of the aks-secure-baseline repo.